tests/CloudEvents/Serializer/JsonSerializerTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace CloudEvents\Tests\CloudEvents\Serializer;
+
+use CloudEvents\Serializer\JsonSerializationOptions;
+use CloudEvents\Serializer\JsonSerializer;
+use CloudEvents\V1\CloudEventInterface;
+use DateTimeImmutable;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversDefaultClass \CloudEvents\Serializer\JsonSerializer
+ */
+class JsonSerializerTest extends TestCase
+{
+    /**
+     * @covers ::serialize
+     */
+    public function testFormat(): void
+    {
+        /** @var CloudEventInterface|MockObject $event */
+        $event = $this->createMock(CloudEventInterface::class);
+        $event->method('getSpecVersion')->willReturn('1.0');
+        $event->method('getId')->willReturn('1234-1234-1234');
+        $event->method('getType')->willReturn('com.example.someevent');
+        $event->method('getDataContentType')->willReturn('application/json');
+        $event->method('getDataSchema')->willReturn('com.example/schema');
+        $event->method('getSubject')->willReturn('larger-context');
+        $event->method('getTime')->willReturn(new DateTimeImmutable('2018-04-05T17:31:00Z'));
+        $event->method('getData')->willReturn(['key' => 'value']);
```

Comment:
Please prefer stubs over mocks.

Reply:
Yup. Definitely used the wrong one there.